addons/mail/static/src/messaging/entity/dialog_manager/dialog_manager.js

```diff
@@ -55,9 +64,10 @@ function DialogManagerFactory({ Entity }) {
             if (!this.allOrdered.includes(dialog)) {
                 return;
             }
+            this.unlink({ dialogs: dialog });
             this.update({
                 _ordered: this._ordered.filter(
-                    _dialog => _dialog === dialog.localId
+                    _dialog => _dialog !== dialog.localId
                 ),
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Filter does not seem to explicitly guaranteed the order? To be safe I think the order code of the dialog should be factorized (it not yet) and called again. It shouldn't need to know what was just unregistered.

Reply:
https://tc39.es/ecma262/#sec-array.prototype.filter the pseudo code seems to say that it keeps the order, no ?

Reply:
no explicit sort after filter: `.filter()` always keep order, which is essential and guaranteed by specs.